phalcon/di.zep

```diff
@@ -354,15 +354,15 @@ class Di implements DiInterface
 	 * var_dump($di["request"]);
 	 *</code>
 	 */
-	public function offsetGet(string! name) -> var
+	public function offsetGet(name) -> var
 	{
 		return this->getShared(name);
 	}
 
 	/**
 	 * Removes a service from the services container using the array syntax
 	 */
-	public function offsetUnset(string! name) -> boolean
+	public function offsetUnset(name) -> boolean
```

Comment:
Do you propose use this as `unset($di[new \stdClass()])`? Why the `name` must not be a string?

Reply:
i am not sure here: i think this is a internal php interface, so this is how they defined it http://php.net/manual/en/arrayobject.offsetunset.php

i could be wrong but this is what i think

Reply:
We do not use `$di[1]` or `$di[false]` or `$di[object]`. The name should be a string

Reply:
php is complaining for all those  `offsetUnset` `offsetGet` `offsetSet` `offsetExists`, the only thing that I was able to put in front was "var" and that is it. Otherwise i am completely agree this has to be a string 100% of the time

Reply:
@Izopi4a Is it fails to build with `string! name`

Reply:
Could you paste the output and PHP version?

Reply:
php -v
PHP 7.2.10-0ubuntu0.18.04.1 (cli) (built: Sep 13 2018 13:45:02) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.10-0ubuntu0.18.04.1, Copyright (c) 1999-2018, by Zend Technologies

Reply:
PHP Fatal error:  Declaration of Phalcon\Di::offsetUnset(string $name): bool must be compatible with ArrayAccess::offsetUnset($offset) in Unknown on line 0